File: src/leetcode/2025/11/Solution169.java
public class Solution169 {
    public int majorityElement(int[] nums) {
        int count = 0;
        Integer candidate = null;

        for (int num : nums) {
            if (count == 0) {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }

    public static void main(String[] args) {
        Solution169 solution = new Solution169();

        // Example 1
        int[] nums1 = { 3, 2, 3 };
        int result1 = solution.majorityElement(nums1);
        System.out.println("Example 1: " + result1);
        // Expected: 3

        // Example 2
        int[] nums2 = { 2, 2, 1, 1, 1, 2, 2 };
        int result2 = solution.majorityElement(nums2);
        System.out.println("Example 2: " + result2);
        // Expected: 2
    }
}

File: src/leetcode/2025/11/Solution169.cpp
#include <vector>
using namespace std;

class Solution
{
public:
    int majorityElement(vector<int> &nums)
    {
        int count = 0;
        int candidate = 0;

        for (int num : nums)
        {
            if (count == 0)
            {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }
};

int main()
{
    // Example 1
    Solution solution;
    vector<int> nums1 = {3, 2, 3};
    int result1 = solution.majorityElement(nums1);
    // Expected Output: 3
    printf("Majority Element: %d\n", result1);

    // Example 2
    vector<int> nums2 = {2, 2, 1, 1, 1, 2, 2};
    int result2 = solution.majorityElement(nums2);
    // Expected Output: 2
    printf("Majority Element: %d\n", result2);

    return 0;
}